test/ibmq/websocket/test_websocket_integration.py

```diff
@@ -26,45 +33,129 @@
-    def test_websockets_simulator(self, qe_token, qe_url):
-        """Test checking status of a job via websockets for a simulator."""
+    def setUp(self, qe_token, qe_url):
         ibmq_factory = IBMQFactory()
-        provider = ibmq_factory.enable_account(qe_token, qe_url)
-        backend = provider.get_backend(simulator=True)
+        self.provider = ibmq_factory.enable_account(qe_token, qe_url)
+        self.sim_backend = self.provider.get_backend(simulator=True)
 
-        qc = transpile(self._qc1, backend=backend)
-        qobj = assemble(qc, backend=backend)
+        # Create a circuit
+        qr = QuantumRegister(1)
+        cr = ClassicalRegister(1)
+        self.qc1 = QuantumCircuit(qr, cr, name='qc1')
+        self.qc1.measure(qr[0], cr[0])
 
-        job = backend.run(qobj)
-        # Manually disable the non-websocket pooling.
-        job._wait_for_final_status = None
-        result = job.result()
+        # Create and run a default Qobj using the simulator.
+        self.circuit = transpile(self.qc1, backend=self.sim_backend)
+        self.qobj = assemble(self.circuit, backend=self.sim_backend, shots=1)
+        self.job = self.sim_backend.run(self.qobj)
+
+    def test_websockets_simulator(self):
+        """Test checking status of a job via websockets for a simulator."""
+        # Manually disable the non-websocket polling.
+        self.job._wait_for_final_status = None
+        result = self.job.result()
 
         self.assertEqual(result.status, 'COMPLETED')
 
     @slow_test
-    @requires_qe_access
-    @requires_new_api_auth
-    def test_websockets_device(self, qe_token, qe_url):
+    def test_websockets_device(self):
         """Test checking status of a job via websockets for a device."""
-        ibmq_factory = IBMQFactory()
-        provider = ibmq_factory.enable_account(qe_token, qe_url)
-        backend = least_busy(provider.backends(simulator=False))
+        backend = least_busy(self.provider.backends(simulator=False))
 
-        qc = transpile(self._qc1, backend=backend)
+        qc = transpile(self.qc1, backend=backend)
         qobj = assemble(qc, backend=backend)
 
         job = backend.run(qobj)
-        # Manually disable the non-websocket pooling.
+        # Manually disable the non-websocket polling.
         job._wait_for_final_status = None
         result = job.result()
 
         self.assertEqual(result.status, 'COMPLETED')
+
+    def test_websockets_job_final_state(self):
+        """Test checking status of a job in a final state via websockets."""
+        # Cancel the job to put it in a final (cancelled) state.
+        self.job.cancel()
+        self.job._wait_for_completion()
+        self.assertIs(self.job._status, JobStatus.CANCELLED)
+
+    def test_websockets_retry_bad_url(self):
+        """Test http retry after websocket error due to an invalid URL."""
+        # Use fake websocket address.
+        self.job._api.client_ws.websocket_url = 'wss://wss.wayne-enterprises.com'
```

Comment:
I like the theme, but in this case can you please use `localhost` or similar, to avoid extra network calls?